src/warehouses/components/WarehouseDetailsPage/WarehouseDetailsPage.tsx

```diff
@@ -8,32 +8,37 @@ import PageHeader from "@saleor/components/PageHeader";
 import Savebar from "@saleor/components/Savebar";
 import { ShopInfo_shop_countries } from "@saleor/components/Shop/types/ShopInfo";
 import { AddressTypeInput } from "@saleor/customers/types";
+import { WarehouseDetailsFragment } from "@saleor/fragments/types/WarehouseDetailsFragment";
 import { WarehouseErrorFragment } from "@saleor/fragments/types/WarehouseErrorFragment";
 import useAddressValidation from "@saleor/hooks/useAddressValidation";
 import { SubmitPromise } from "@saleor/hooks/useForm";
 import useStateFromProps from "@saleor/hooks/useStateFromProps";
 import { sectionNames } from "@saleor/intl";
 import { Backlink } from "@saleor/macaw-ui";
 import { findValueInEnum, maybe } from "@saleor/misc";
-import { CountryCode } from "@saleor/types/globalTypes";
+import {
+  CountryCode,
+  WarehouseClickAndCollectOptionEnum
+} from "@saleor/types/globalTypes";
 import createSingleAutocompleteSelectHandler from "@saleor/utils/handlers/singleAutocompleteSelectChangeHandler";
 import { mapCountriesToChoices, mapEdgesToItems } from "@saleor/utils/maps";
 import React from "react";
 import { FormattedMessage, useIntl } from "react-intl";
 
-import { WarehouseDetails_warehouse } from "../../types/WarehouseDetails";
 import WarehouseInfo from "../WarehouseInfo";
-import WarehouseZones from "../WarehouseZones";
+import WarehouseSettings from "../WarehouseSettings";
 
 export interface WarehouseDetailsPageFormData extends AddressTypeInput {
   name: string;
+  isPrivate: string;
```

Comment:
Why is this a string and not a boolean type?

Reply:
Since the radio button value can `string or number` it is easier to hold it as a string and cast it to `boolean` only once, during Warehouse update.

Reply:
I think we shouldn't make our form data types dependent on the type of inputs we use to change them. After all, these can change. Also, it's confusing since in the code for this prop we see stuff like `=== "true"` , whereas in other places in our codebase when we use booleans we actually use them as booleans, not strings. That's creating double standards, which I don't think are either helpful nor necessary. I think a good solution here would be a component `BoolRadioGroupField`, that takes is a wrapper for RadioGroupField that parses those values when passing to the radios and returns those values as booleans when changed. Or at least a handler function, @jwm0?

Reply:
@kuchichan it should be the other way around. It will be easier to have it as a boolean inside form as this will allow to use traditional bool checks and make it less prone to errors in future. Where you should be casting it is the actual input itself. Example:
```
// WarehouseSettings.tsx
const booleanRadioHandler = ({ target: { name, value } }) => {
  setData({ [name]: value === "true" });
}

<Radio value={data.isPrivate.toString()} onChange={booleanRadioHandler} />
```

Reply:
@mmarkusik I'd rather stick to @jwm0 advice, if you don't mind. What is more, it's done ;)